sys/include/xtimer.h

```diff
@@ -389,6 +393,75 @@ int xtimer_msg_receive_timeout64(msg_t *msg, uint64_t us);
 #define XTIMER_USLEEP_UNTIL_OVERHEAD 10
 #endif
 
+#ifndef XTIMER_HZ
+/**
+ * @brief Tick rate of the underlying hardware timer.
+ *
+ * This value is used to convert between hardware ticks and microseconds.
+ *
+ * This is supposed to be defined per-device in e.g., periph_conf.h.
+ */
+#define XTIMER_HZ (1000000ul)
+#endif
+
+/* Some optimizations for common timer frequencies */
+#if XTIMER_HZ == (1000000ul)
+/* This is the most straightforward as the xtimer API is based around
+ * microseconds for representing time values. */
+#define _xtimer_us_to_ticks(us)    (us) /* no-op */
+#define _xtimer_ticks_to_us(ticks) (ticks) /* no-op */
+#define _xtimer_us_to_ticks64(us)    (us) /* no-op */
+#define _xtimer_ticks_to_us64(ticks) (ticks) /* no-op */
+
+#elif XTIMER_HZ == (32768ul)
+/* This is a common frequency for RTC crystals. We use the fact that the
+ * greatest common divisor between 32768 and 1000000 is 64, so instead of
+ * multiplying by the fraction (32768 / 1000000), we will instead use
+ * (512 / 15625), which reduces the truncation caused by the integer widths */
+
+inline static uint32_t _xtimer_us_to_ticks(uint32_t us) {
+    /* return (usec * 512) / 15625; */
+    /*
+     * Optimization to avoid division instructions or division libraries.
+     * This code gives equivalent machine code on ARMv7-M (Cortex-M3, M4) as
+     * generated by compiling the above code line.
+     * at -O2 or higher with GCC 4.9. Cortex-M4 with optimization level -Os or
+     * lower does not automatically perform this optimization.
+     * Cortex-M0 (ARMv6-M) does not have hardware division and will insert a
+     * call to the helper function __aeabi_uidiv which is quite complex and
+     * slow, and uses an additional 200 bytes of ROM compared to the helper
+     * function required for multiplying two 32 bit integers into a 64 bit result.
+     */
+    return ((uint64_t)(us << 9) * 0x431bde83ul) >> (12 + 32);
+}
+
+inline static uint64_t _xtimer_us_to_ticks64(uint64_t us) {
+    /* Same principle as the 32 bit version. */
+    /* Magic number taken from a disassembly of GCC's output on x86_64 for the
+     * below line:
+     *
+     * return (usec * 512) / 15625;
+     */
+    /* Saves around 1400 bytes of ROM on Cortex-M platforms (both ARMv6 and
+     * ARMv7) from avoiding linking against __aeabi_uldivmod and related helpers
+     */
+    return ((uint64_t)(us << 9) * 0x431bde82d7b634dbull) >> 12;
+}
+
+inline static uint32_t _xtimer_ticks_to_us(uint32_t ticks) {
+    /* Using 64 bit multiplication to avoid truncating the top 9 bits */
+    uint64_t us = (uint64_t)ticks * 15625ul;
+    return us >> 9;
```

Comment:
For `_xtimer_us_to_ticks()` you commented, what the actual meaning of the calculation is. Can you please do so here, too?